share/ruby-install/truffleruby/functions.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Check platform
+case $(uname) in
+	Linux) platform=linux ;;
+	Darwin) platform=macos ;;
+	*) fail "unknown platform $(uname)" ;;
+esac
+
+# Check architecture
+case $(uname -m) in
+	x86_64) arch=amd64 ;;
+	*) fail "unknown architecture $(uname -m)" ;;
+esac
```

Comment:
Should we extract these into `util.sh`?

Reply:
As you prefer, both are fine for me.
The resulting values might not be suitable for all usages though.
For instance, I can imagine some usages could expect `darwin` instead of `macos` for macOS.

Reply:
Can we get away with using `OSTYPE` and `MACHTYPE` since it's bash?

```bash
case "$OSTYPE" in
	linux*) platform=linux ;;
	darwin*) platform=macos ;;
	*) fail "incompatible platform $OSTYPE" ;;
esac

case "$MACHTYPE" in
	x86_64) arch=amd64 ;;
	*) fail "incompatible architecture $MACHTYPE" ;;
esac
```

Reply:
I guess so, but I think the output of `uname` is more standard and does not vary as much (e.g., `MACHTYPE` is `x86_64-redhat-linux-gnu` on Bash, `x86_64` on ZSH and `OSTYPE` is `linux-gnu`).
`uname` is also what TruffleRuby uses upstream so we know it works for all our supported platforms.

Reply:
@eregon That makes sense.

I wonder if it's premature to extract this to `util.sh`, even though that ultimately seems like a good idea. Maglev was using `platform` and `arch`, but with it gone there's nothing else currently using them - and as pointed out, the desired `macos` and `amd64` values are domain specific.